python/lsst/meas/extensions/ngmix/LMSimpleShape.py

```diff
@@ -0,0 +1,238 @@
+            raise RuntimeError("Threshold value results in zero Footprints for object")
+        fp = fpSet.getFootprints()[0]
+        x = fp.getCentroid().getX() - exposure.getXY0().getX()
+        y = fp.getCentroid().getY() - exposure.getXY0().getY()
+        xx = fp.getShape().getIxx()
+        yy = fp.getShape().getIyy()
+        xy = fp.getShape().getIxy()
+        #    set the guess as [cen1, cen2, g1, g2, T, flux]
+        guess = (x, y, (xx-yy)/(xx+yy), 2*xy/(xx+yy), xx+yy, galArray[x, y])
+
+        #   Now run the shape algorithm, using the config parameters
+        lmPars = {'maxfev': self.config.maxfev, 'ftol': self.config.ftol, 'xtol': self.config.xtol}
+        fitter = LMSimple(obs, self.config.model, lm_pars=lmPars)
+        fitter.go(guess=guess)
+        res = fitter.get_result()
+
+        #   Set the results, including the fit info returned by EMRunner
+        #   We only know about two EM errors.  Anything else is thrown as "unknown".
+        if res['flags'] != 0:
+            if res['flags'] & LM_SINGULAR_MATRIX:
+                raise MeasurementError(self.flagHandler.getDefinition(
+                    SingleFrameNgmixPlugin.ErrEnum.flag_singularMatrix).doc,
+                    SingleFrameNgmixPlugin.ErrEnum.flag_singularMatrix)
+            if res['flags'] & LM_NEG_COV_EIG:
+                raise MeasurementError(
+                    self.flagHandler.getDefinition(SingleFrameNgmixPlugin.ErrEnum.flag_negativeCovEig).doc,
+                    SingleFrameNgmixPlugin.ErrEnum.flag_negativeCovEig)
+            if res['flags'] & LM_NEG_COV_DIAG:
+                raise MeasurementError(
+                    self.flagHandler.getDefinition(SingleFrameNgmixPlugin.ErrEnum.flag_negativeCovDiag).doc,
+                    SingleFrameNgmixPlugin.ErrEnum.flag_negativeCovDiag)
+            if res['flags'] & EIG_NOTFINITE:
+                raise MeasurementError(
+                    self.flagHandler.getDefinition(SingleFrameNgmixPlugin.ErrEnum.flag_eigNotFinite).doc,
+                    SingleFrameNgmixPlugin.ErrEnum.flag_eigNotFinite)
+            if res['flags'] & LM_FUNC_NOTFINITE:
+                raise MeasurementError(
+                    self.flagHandler.getDefinition(SingleFrameNgmixPlugin.ErrEnum.flag_funcNotFinite).doc,
+                    SingleFrameNgmixPlugin.ErrEnum.flag_funcNotFinite)
+            if res['flags'] & LM_DIV_ZERO:
+                raise MeasurementError(
+                    self.flagHandler.getDefinition(SingleFrameNgmixPlugin.ErrEnum.flag_divZero).doc,
+                    SingleFrameNgmixPlugin.ErrEnum.flag_divZero)
+            if res['nfev'] >= self.config.maxfev:
+                raise MeasurementError(
+                    self.flagHandler.getDefinition(SingleFrameNgmixPlugin.ErrEnum.flag_maxFev).doc,
+                    SingleFrameNgmixPlugin.ErrEnum.flag_maxFev)
+            #   Unknown error, but there should be an errmsg set by ngmix
+            raise RuntimeError(res['errmsg'])
```

Comment:
I think I am OK with having completely unnumbered errors bring the whole task.run down.  That was my intention.  As far as you question about errors is concerned, I think that RuntimeError turns into a FatalAlgorithmError, which is in FATAL_EXCEPTIONS in baseMeasurement.  This is not caught by the framework.